src/Driver/ResultStatement.php

```diff
@@ -37,40 +37,83 @@ public function columnCount() : int;
     public function rowCount() : int;
 
     /**
-     * Sets the fetch mode to use while iterating this statement.
+     * Returns the next row of a result set as a numeric array or FALSE if there are no more rows.
      *
-     * @param int $fetchMode Controls how the next row will be returned to the caller.
-     *                       The value must be one of the {@link \Doctrine\DBAL\FetchMode} constants.
+     * @return array<int,mixed>|false
```

Comment:
I think PDO returns `false` in its API because it makes methods usable with a `while` statement.
If we are being strict though, it implies casting an array to a boolean in most cases. How about we use `null` instead here? `null` means absence of data, so I think it makes more sense from a semantic PoV, and as a bonus, we can have a return type declaration here.

Reply:
This is a valid point. Although `null` is indeed more semantically correct, the reason why PDO uses `false` to indicate the end of result is because `$stmt->fetch(PDO::FETCH_COLUMN)` may return `null` as a value.

Regardless of whether return `?array` or `array|false` in `fetchNumeric()` and `fetchAssociative()`, we'll have to return `mixe|false` in `fetchOne()`. The fact that all three can be used in a `while ()` statement makes me believe that consistency is more important than purity in each individual case.

What do you think?

Reply:
It makes sense, let's keep it the way you wrote it 👍